File: 2604-minimum-operations-to-make-array-equal-ii/minimum-operations-to-make-array-equal-ii.py
class Solution:
    def minOperations(self, nums1: List[int], nums2: List[int], k: int) -> int:
        if not k: return 0 if nums1 == nums2 else -1
        s,op = 0,0
        for i in range(len(nums1)):
            if abs(nums1[i] - nums2[i])%k != 0:
                return -1
            s += nums1[i] - nums2[i]
            op += abs(nums1[i] - nums2[i])//k
        return op//2 if s == 0 else -1

File: 2604-minimum-operations-to-make-array-equal-ii/minimum-operations-to-make-array-equal-ii.cpp
class Solution {
public:
    long long minOperations(vector<int>& nums1, vector<int>& nums2, int k) {
        if(!k)return (nums1 == nums2 ? 0 : -1);
        long long s = 0 , op = 0;
        for(int i =  0 ; i < nums1.size();i++){
            if(abs(nums1[i] - nums2[i])%k)return -1;
            s+=nums1[i] - nums2[i];
            op+=abs(nums1[i] - nums2[i])/k;
        }
        return !s ? op/2 : -1;

    }
};